zanata-client-commands/src/main/java/org/zanata/client/commands/push/PushCommand.java

```diff
@@ -452,14 +455,25 @@ private static void stripUntranslatedEntriesIfMergeTypeIsNotImport(
         if (!MergeType.IMPORT.name().equalsIgnoreCase(mergeType)) {
             List<TextFlowTarget> originalTargets =
                     translationResources.getTextFlowTargets();
-            Collection<TextFlowTarget> untranslatedEntries = Collections2
-                    .filter(originalTargets,
+            final Predicate<String> blankStringPredicate =
+                    new Predicate<String>() {
+                        @Override
+                        public boolean apply(String input) {
+                            return Strings.isNullOrEmpty(input);
+                        }
+                    };
+
+            Collection<TextFlowTarget> untranslatedEntries =
+                    filter(originalTargets,
                             new Predicate<TextFlowTarget>() {
                                 @Override
                                 public boolean apply(
                                         TextFlowTarget input) {
-                                    return input == null ||
-                                            input.getState().isUntranslated();
+                                    // it's unsafe to rely on content state (plural entries)
+                                    return input == null
+                                            || input.getContents().isEmpty()
+                                            || all(input.getContents(),
+                                                    blankStringPredicate);
```

Comment:
It's not reliable when dealing with plural entries. We treat it as untranslated if not all plurals have value. Therefore the original implementation will miss some partial translations